CMakeLists.txt

```diff
@@ -177,16 +177,21 @@ if (CMAKE_GENERATOR MATCHES "Xcode")
     # So we need to build from source, and then add all the other settings missing due to lack of lslobj
     add_library(lsl SHARED ${lslobj_sources})
     target_link_libraries(lsl PRIVATE lslboost)
-    target_include_directories(lsl
-	PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>)
-    target_compile_definitions(lsl
-	PRIVATE LIBLSL_EXPORTS $<$<PLATFORM_ID:Windows>:_CRT_SECURE_NO_WARNINGS>
-	INTERFACE LSLNOAUTOLINK # don't use #pragma(lib) in CMake builds
-    )
 else()
     add_library(lsl SHARED)
-    target_link_libraries(lsl PUBLIC lslobj PRIVATE lslboost)
+    target_link_libraries(lsl
+		PUBLIC lslobj
+		PRIVATE lslboost)
 endif()
+target_include_directories(lsl
+	PUBLIC
+		$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>
+		$<INSTALL_INTERFACE:include>
+)
+target_compile_definitions(lsl
+	PRIVATE LIBLSL_EXPORTS $<$<PLATFORM_ID:Windows>:_CRT_SECURE_NO_WARNINGS>
+	INTERFACE LSLNOAUTOLINK # don't use #pragma(lib) in CMake builds
```

Comment:
The private definitions don't get exported and `LSLNOAUTOLINK` only affects MSVC, so it could stay behind the `if`